javaslang/src/test/java/javaslang/control/TryTest.java

```diff
@@ -456,6 +460,27 @@ public void shouldConvertFailureToEitherLeftSupplier() {
         assertThat(failure().toEither(() -> "test").isLeft()).isTrue();
     }
 
+    // -- toCompletableFuture
+
+    @Test
+    public void shouldConvertSuccessToCompletableFuture()  {
+        CompletableFuture<String> future = success().toCompletableFuture();
+        assertThat(future.isDone());
+        assertThat(Try.of(future::get).get()).isEqualTo(success().get());
+    }
+
+    @Test
+    public void shouldConvertFailureToFailedCompletableFuture() {
+
+        CompletableFuture<Object> future = failure().toCompletableFuture();
+        assertThat(future.isDone());
+        assertThat(future.isCompletedExceptionally());
+        assertThatThrownBy(future::get)
+                .isExactlyInstanceOf(ExecutionException.class)
+                .hasCauseExactlyInstanceOf(NonFatalException.class)
```

Comment:
Yes, or unbox a NonFatalException in the default `toCompletableFuture` impl.

Unboxing it is perfectly ok because there is no public API that allows a call-site to construct a NonFatalException outside of a Try. Please find the suggested impl above...